Sources/Apodini/Semantic Model Builder/Model/Endpoint.swift

```diff
@@ -3,34 +3,18 @@
 //
 import Foundation
 
-/// Models a single Endpoint which is identified by its PathComponents and its operation
-public protocol AnyEndpoint: Blackboard, CustomStringConvertible {
-    /// All `@Parameter` `RequestInjectable`s that are used inside handling `Component`
-    var parameters: [AnyEndpointParameter] { get }
-
-    var absolutePath: [EndpointPath] { get }
-
-    /// Returns the `RelationshipDestination` (with Operation equal to `operation`) for the given Endpoint
-    var selfRelationship: RelationshipDestination { get }
-
-    /// Creates a set of `RelationshipDestination` which ensures that relationship names
-    /// are unique for a every `Operation`
-    /// - Returns: The set of uniquely named relationship destinations.
-    func relationships() -> Set<RelationshipDestination>
-
-    /// Creates a set of `RelationshipDestination` which ensures that relationship names
-    /// are unique (for all collected destination for a given `Operation`)
-    /// - Parameter operation: The `Operation` of the Relationship destination to create a unique set for.
-    /// - Returns: The set of uniquely named relationship destinations.
-    func relationships(for operation: Operation) -> Set<RelationshipDestination>
-
-    /// Returns the special "self" Relationship for all `Operation`s.
-    func selfRelationships() -> Set<RelationshipDestination>
-
-    /// Returns the special "self" Relationship for a given `Operation`
-    /// - Parameter for: The `Operation` for the desired destination.
-    func selfRelationship(for: Operation) -> RelationshipDestination?
+/// A ``ParameterCollection`` provides random access to an endpoint's ``EndpointParameter``s.
+public protocol ParameterCollection {
```

Comment:
Looking at the name of the protocol `ParameterCollection`, one would expect to have a parameters collection requirement. I would suggest to adjust it with a `var parameters: [AnyEndpointParameter] { get }` and provide a default implementation for `findParameter(for:)`

Reply:
Seems reasonable👍🏻